### PR TITLE
Devise Installation

### PR DESCRIPTION
**Pull request by HFG43**
Hi there! I'm sending this pull request for you to review my code. Thanks in advance for your feedback!

Milestones 🎯:
📌 Work on Pair Programming
📌 Add devise Gem
📌 Add letter oppener gem
📌 Add user registration & login
📌 Ask for confirmation mail
📌 Add reset password option

🎖️ General requirements:
Gitflow
No linter errors
Ensure the use of best practices and Ruby style

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.password_field :password, placeholder: "Password", autocomplete: "new-password" %>
+  </div>
+
+  <div class="field form_input ">
+    <%= f.password_field :password_confirmation, placeholder: "Confirm password", autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions form_input">
+    <%= f.submit "Sign up" , class: "button"%>
+  </div>
```
Comment (Strangeal) [inline app/views/devise/registrations/new.html.erb:28]: - [ ] Great job on your implementation so far! 👏 However, I'd like to highlight a crucial aspect mentioned in the project requirements. It's important to ensure that the `confirmation` module is implemented in such a way that when a user creates a new account, they are prompted to confirm their email by clicking on a generated link before gaining access to their account. This step is essential for enhancing the security and usability of the application. Please make sure to address this requirement in your implementation. 

Happy coding! 😊🚀🍀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @HFG43 and @ahmedeid6842, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Pixel Perfect Design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HFG43** [comment]
@Strangeal Thanks for your time and feedback!! We will work on these adjustments!

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="field form_input">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in", class: "button" %>
+  </div>
+<% end %>
```
Comment (Strangeal) [inline app/views/devise/sessions/new.html.erb:23]: - [ ] Great effort so far! 👏 However, I noticed that the suggested change regarding redirecting a user to the home page after a successful sign-in hasn't been implemented yet. It's crucial for user experience that, upon signing in using their email and password, they should be seamlessly redirected to the home page of your application. 🏠 This not only enhances user navigation but also provides a smoother flow within your app. Please consider implementing this change to improve the overall user experience. 

Happy coding! 😊🚀🍀

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (Strangeal) [inline app/views/devise/registrations/new.html.erb:28]: - [ ] Great effort in implementing the `confirmation` feature! 👏 However, there's a minor flow issue that needs addressing. 

Currently, after a user has successfully created an account, they are redirected to the home page of the app. This isn't aligned with the intended flow. 

### Here's the recommended flow:

1. On a successful `sign_up`, users should be directed to a dedicated `UI` that asks them to `confirm` their email. This step is crucial for email verification.

2. After users have successfully confirmed their email, that's when you can redirect them to the home page or the login page for them to sign into their account.

This adjustment ensures a smoother and more secure onboarding experience for users.
Happy coding! 😊🚀🍀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @HFG43 and @ahmedeid6842, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Highlights
- Reset password `UI` ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (HFG43) [reply app/views/devise/registrations/new.html.erb:28]: @Strangeal Hi Justice, please review this video where I show that this redirection is already in place https://www.loom.com/share/713ef1b2e1b7409eb3b1a9a77c188d97?sid=8582e224-d740-4ea8-a3f5-515653d91a2c

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (HFG43) [reply app/views/devise/sessions/new.html.erb:23]: @Strangeal Hi there! Please, following the previous video, please find here one that shows that after the correct sign in is being redirect to the home path. 
https://www.loom.com/share/0bf1379b3adf481d9577d47ebf95f13d?sid=e5fea012-4278-4909-ad08-fdebaa2c3602

**Comment by nicupop729** [comment]
This PR is under a Second Opinion Review.

Please wait for the review.

**Review by nicupop729 — COMMENTED** [review]
Hello @HFG43 & @ahmedeid6842 👋,

_This is an additional review as a result of the second opinion request._

In my opinion, all project requirements are implemented and the project is ready to be approved.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Comment by HFG43** [comment]
@nicupop729 Hi Nicolae! Thank you very much for your time and feedback!!

**Review by KDlamini — APPROVED** [review]
### Hi team @HFG43, @ahmedeid6842 🙋‍♀️,  

Excellent work! 💯 
You have done an outstanding job adding the required changes for the Devise authentication to the Blog app. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞 

✔️  Good PR title and description
✔️  Well documented and professional README
✔️  A user can register and login with a combination of email and password
✔️  A user can reset the password in the application
✔️  Kudos! for using commit types to write meaningful commit messages 🚀 

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HFG43** [comment]
@KDlamini Hi Simo!! Many thanks for your feedback!

**Pull request merged**